awx/main/tests/unit/test_tasks.py

```diff
@@ -1008,7 +1010,8 @@ def test_openstack_credentials(self, private_data_dir, job, mock_me):
         credential.credential_type.inject_credential(credential, env, {}, [], private_data_dir)
 
         config_loc = to_host_path(env['OS_CLIENT_CONFIG_FILE'], private_data_dir)
-        shade_config = open(config_loc, 'r').read()
+        with open(config_loc, 'r') as f:
+            shade_config = f.read()
```

Comment:
@AlanCoding here's an example of what I meant @ https://github.com/ansible/awx/pull/15620#discussion_r1837195000
```suggestion
        shade_config = Path(config_loc).read_text()
```